test/skimNtuple2018_HHbtag.cpp

```diff
@@ -2090,32 +2098,63 @@ int main (int argc, char** argv)
-      //theSmallTree.m_tauH_SVFIT_mass_METdown = (isMC ? theBigTree.SVfitMassMETDown->at (chosenTauPair) : theSmallTree.m_tauH_SVFIT_mass);
-      // in case the SVFIT mass is calculated
-      if (theBigTree.SVfitMass->at (chosenTauPair) > -900.)
+
+      if (doSmearing)
+      {
+        // Smear MET
+        float METx = theBigTree.METx->at(chosenTauPair);
+        float METy = theBigTree.METy->at(chosenTauPair);
+        TVector2 metSmeared = getShiftedMET_smear(METx, METy, theBigTree, Smearer);
+        vMET = metSmeared;
+        tlv_MET.SetPxPyPzE(metSmeared.Px(), metSmeared.Py(), 0, std::hypot(metSmeared.Px(), metSmeared.Py()));
+
+        // Recompute SVfit after smearing the MET
+        TMatrixD metcov (2, 2) ;
+        metcov (0,0) = theBigTree.MET_cov00->at (chosenTauPair) ;
+        metcov (1,0) = theBigTree.MET_cov10->at (chosenTauPair) ;
+        metcov (0,1) = theBigTree.MET_cov01->at (chosenTauPair) ;
+        metcov (1,1) = theBigTree.MET_cov11->at (chosenTauPair) ;
+
+        SVfitKLUBinterface algo_central(0, tlv_firstLepton, tlv_secondLepton, tlv_MET, metcov, pType, theSmallTree.m_dau1_decayMode, theSmallTree.m_dau2_decayMode);
+        std::vector<double> svfitResSmeared = algo_central.FitAndGetResult();
+        theSmallTree.m_tauH_SVFIT_mass = svfitResSmeared.at(3);
+        if (theSmallTree.m_tauH_SVFIT_mass > 0)
         {
-          theSmallTree.m_tauH_SVFIT_pt     = theBigTree.SVfit_pt->at  (chosenTauPair) ;
-          theSmallTree.m_tauH_SVFIT_eta    = theBigTree.SVfit_eta->at (chosenTauPair) ;
-          theSmallTree.m_tauH_SVFIT_phi    = theBigTree.SVfit_phi->at (chosenTauPair) ;
-          theSmallTree.m_tauH_SVFIT_METphi = theBigTree.SVfit_fitMETPhi->at (chosenTauPair) ;
-          theSmallTree.m_tauH_SVFIT_METrho = theBigTree.SVfit_fitMETRho->at (chosenTauPair) ;
-
-          tlv_tauH_SVFIT.SetPtEtaPhiM (
-                    theBigTree.SVfit_pt->at (chosenTauPair),
-                    theBigTree.SVfit_eta->at (chosenTauPair),
-                    theBigTree.SVfit_phi->at (chosenTauPair),
-                    theBigTree.SVfitMass->at (chosenTauPair)
-                    ) ;
-
-          theSmallTree.m_tauHsvfitMet_deltaPhi = deltaPhi (theBigTree.metphi, tlv_tauH_SVFIT.Phi ()) ;
+          theSmallTree.m_tauH_SVFIT_pt   = svfitResSmeared.at(0);
+          theSmallTree.m_tauH_SVFIT_eta  = svfitResSmeared.at(1);
+          theSmallTree.m_tauH_SVFIT_phi  = svfitResSmeared.at(2);
+          tlv_tauH_SVFIT.SetPtEtaPhiM(svfitResSmeared.at(0), svfitResSmeared.at(1), svfitResSmeared.at(2), svfitResSmeared.at(3));
+
+          theSmallTree.m_tauHsvfitMet_deltaPhi = deltaPhi (vMET.Phi(), tlv_tauH_SVFIT.Phi ()) ;
           theSmallTree.m_ditau_deltaR_per_tauHsvfitpt = tlv_firstLepton.DeltaR(tlv_secondLepton) * tlv_tauH_SVFIT.Pt();
         }
+      }
 
-    if (DEBUG)
+      if (theBigTree.SVfitMass->at (chosenTauPair) > -900.)
```

Comment:
Bug! this should be:
`if (!doSmearing && ...)`
Right @camendola and @dzuolo ?

Reply:
Yes!

Reply:
addressed in bf84663